substrate-node/pallets/pallet-tft-bridge/src/types.rs

```diff
@@ -17,9 +30,10 @@ pub struct MintTransaction<AccountId, BlockNumber> {
 // TF Chain -> Stellar burn transaction
 // Transaction is ready when (number of validators / 2) + 1 signatures are present
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Decode, Default, Debug, TypeInfo)]
-pub struct BurnTransaction<BlockNumber> {
+pub struct BurnTransaction<AccountId, BlockNumber> {
     pub block: BlockNumber,
     pub amount: u64,
+    pub source: Option<AccountId>,
```

Comment:
Because the previous transactions do not have a source account, and there is no suitable value to assign, I decided to set the source of all previous transactions to none. Therefore, OptionQuery is necessary here.”

Reply:
I see
Not ideal but with the previous transactions case this is a way to handle it as you said.
Another way would have been to create a new `OptionQuery` storage map associating a `tx_id` to a source account.
Something like this:

~~~rust
    pub type BurnTransactionSource<T: Config> = StorageMap<
        _,
        Blake2_128Concat,
        u64,
        T::AccountId,
        OptionQuery,
    >;
~~~

Moreover, this would not have required any migration process.
This is the solution we tended to chose for the last similar situations, to get more flexibility (if you want to revert/discard some design choice, just nuke storage instead of doing migration again)
What do you think ?

Reply:
Indeed, this is another way to approach this. However, I’m not sure which one is cleaner.
After considering the approach you mentioned, I’m leaning toward the current approach and I can explain why.

- If the only motive here is to avoid migration, I would say there is no need as the migration here is very straightforward (although this may not be the case in every scenario).
- When you think about the data structure, it is obvious that this information is critical to be included in the transaction, and separating it into different storage maps seems less clean in this case.
- It is safe to assume that we won’t need to remove the source information from the transactions later.
- The migration will be a one-time operation. With the other approach, you can skip the migration, but for querying the transaction info, you will need to query two storage keys. This translates to two read operations versus one, and for external apps, this will translate to two RPC calls versus one call only at the moment. This seems like an unnecessary cost to pay.

Reply:
Ok, let's go for it.
What is important to me is that we have in mind that migration need to be avoided as much as possible.
Even if cleaner at the end we lose a lot of energy on it and exposes us to "unnecessary" risks.
But let s keep it like this since work is done and you get you first migration!